sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -311,6 +344,9 @@ object FileStreamSource {
   /** Timestamp for file modification time, in ms since January 1, 1970 UTC. */
   type Timestamp = Long
 
+  val DISCARD_UNSEEN_FILES_RATIO = 0.2
+  val MAX_CACHED_UNSEEN_FILES = 10000
```

Comment:
Any reason for keeping these 2 parameters instead of making it configurable? Is it to detail to expose to the end-user?

Reply:
I just wanted to avoid Spark configuration be "airplane control panel" - end users already have bunch of things to tune. It's completely OK to make them be configurable, if we found the case the default value won't work.